temporalcli/client.go

```diff
@@ -155,10 +156,19 @@ func clientIdentity() string {
 	return "temporal-cli:" + username + "@" + hostname
 }
 
-type stringMapHeadersProvider map[string]string
+type headersProvider struct {
+	setByArgs map[string]string
+}
 
-func (s stringMapHeadersProvider) GetHeaders(context.Context) (map[string]string, error) {
-	return s, nil
+func (h *headersProvider) GetHeaders(context.Context) (map[string]string, error) {
+	retme := make(map[string]string)
+	maps.Copy(retme, h.setByArgs)
+	if _, ok := retme["Authorization"]; !ok {
+		if authHeader := os.Getenv(`TEMPORAL_CLI_AUTHORIZATION_TOKEN`); authHeader != "" {
+			retme["Authorization"] = authHeader
+		}
```

Comment:
You could simplify this with a `newStringMapHeadersProvider` function that pre-fills the Auth header, while retaining all the old logic

That'd also allow folks to override the environment variable with an argument

Reply:
Goo point. I was thinking it might be nice to change the env var at runtime but... then I remembered that's not really a thing.